pkg/agent/watchers.go

```diff
@@ -95,10 +96,12 @@ func (w *AmbassadorWatcher) EnsureStarted(ctx context.Context) {
 type SIWatcher struct {
 	cond           *sync.Cond
 	serviceWatcher *k8sapi.Watcher[*kates.Service]
+	ingressWatcher *k8sapi.Watcher[*snapshot.Ingress]
 }
 
 func NewSIWatcher(clientset *kubernetes.Clientset) *SIWatcher {
 	coreClient := clientset.CoreV1().RESTClient()
+	betaClient := clientset.ExtensionsV1beta1().RESTClient()
```

Comment:
Ingress is now in the `networking.k8s.io/v1` api group as of v1.19 and actually produces an error as of v1.22:
https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122

Reply:
snapshot.k8.ingress is typed as `types_ext_v1beta1.Ingress`, but the fields inside are the same so it should still work I think, good catch